src/main/java/com/endava/internship/collections/StudentMap.java

```diff
@@ -1,72 +1,396 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Consumer;
 
 public class StudentMap implements Map<Student, Integer> {
+
+    private final int DEFAULT_CAPACITY = 1 << 4;
+    private final int MAXIMUM_CAPACITY = 1 << 30;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    final class Node {
+        final int hash;
+        final Student key;
+        Integer value;
+        Node next;
+
+        Node(int hash, Student key, Integer value, Node next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+        public final Student getKey() {
+            return key;
+        }
+        public final Integer getVal() {
+            return value;
+        }
+        public final int getHash() {
+            return hash;
+        }
+        public final Node getNext() {
+            return next;
+        }
+        public void setVal(Integer value) {
+            this.value = value;
+        }
+        public void setNext(Node next) {
+            this.next = next;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if(obj instanceof Node) {
+                Node another = ((Node) obj);
+                return this.getHash() == another.getHash() &&                   // если равны по hash коду
+                        (this.getKey() == null && another.getKey() == null) ||  // проверить, не являются оба null
+                        this.getKey().equals(another.getKey());                 // или не содержат обинаковые значения
+            }
+            return false;
+        }
+    }
+    private Node[] table;
```

Comment:
move them to the top of class, it is not beatiful to see them in the midle of the class